lib/samples/control_annotation_sublayer_visibility/control_annotation_sublayer_visibility.dart

```diff
@@ -0,0 +1,231 @@
+                    }
+                  },
+                ),
+              ],
+            ),
+          if (_closedLabel != null)
+            Row(
+              children: [
+                Text(
+                  _closedLabel!,
+                ),
+                const Spacer(),
+                Switch(
+                  value: _closedSublayer!.isVisible,
+                  onChanged: (value) {
+                    // Set the visibility of the closed sub layer.
+                    if (_closedSublayer != null) {
+                      setState(() => _closedSublayer!.isVisible = value);
+                    }
+                  },
+                ),
+              ],
+            ),
+          Text(
+            _currentScaleLabel ?? '',
+          ),
+        ],
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    await downloadSampleData(['b87307dcfb26411eb2e92e1627cb615b']);
+    final appDir = await getApplicationDocumentsDirectory();
+
+    // Load the mobile map package.
+    final mmpkFile = File('${appDir.absolute.path}/GasDeviceAnno.mmpk');
+    // Mobile map package that contains annotation layers.
+    final mmpk = MobileMapPackage.withFileUri(mmpkFile.uri);
+    await mmpk.load();
+
+    if (mmpk.maps.isNotEmpty) {
+      // Get the first map in the mobile map package and set to the map view.
+      _mapViewController.arcGISMap = mmpk.maps.first;
+    }
+
+    if (_mapViewController.arcGISMap != null) {
+      // Get the annotation layer from the MapView operational layers.
+      final annotationLayer = _mapViewController.arcGISMap!.operationalLayers
+          .whereType<AnnotationLayer>()
+          .first;
+
+      // Load the annotation layer.
+      await annotationLayer.load();
+
+      // Get the annotation sub layers.
+      _closedSublayer =
+          annotationLayer.subLayerContents[0] as AnnotationSublayer;
+      _openSublayer = annotationLayer.subLayerContents[1] as AnnotationSublayer;
+
+      if (_closedSublayer != null) {
+        // Set the closed label content.
+        _closedLabel = _closedSublayer!.name;
+      }
+
+      if (_openSublayer != null) {
+        // Set the open label content.
+        _openLabel =
+            '${_openSublayer!.name} (1:${_openSublayer!.maxScale.toInt()} - 1:${_openSublayer!.minScale.toInt()})';
+
+        // Add event handler for changing the text color to indicate whether the "open" sublayer is visible at the current scale.
+        _mapViewController.onViewpointChanged.listen((_) {
+          // Check if the sublayer is visible at the current map scale.
+          if (_openSublayer!.isVisibleAtScale(_mapViewController.scale)) {
+            _openLabelColor = Colors.purple;
+          } else {
+            _openLabelColor = null;
+          }
+          // Set the current map scale text.
+          _currentScaleLabel =
+              'Current map scale: 1:${_mapViewController.scale.toInt()}';
+
+          setState(() {});
```

Comment:
This works, but `setState(() {});` is frowned upon because it doesn't tell us what has changed. Ideally there'd be a `setState` wrapped around each line of code that impacts the widget.